server/etcdserver/util.go

```diff
@@ -84,6 +84,9 @@ func longestConnected(tp rafthttp.Transporter, membs []types.ID) (types.ID, bool
 type notifier struct {
 	c   chan struct{}
 	err error
+
+	// pass some values in the notifier
+	uint64Val uint64
```

Comment:
```suggestion
	readIndex uint64
```

It's an unexported pkg local struct, I think explicitly naming it is better than generic.

Reply:
notifier was meant to be generic. Not sure if we should add fields here.